.github/actions/artifact_upload_action/action.yml

```diff
@@ -0,0 +1,52 @@
+name: "Artifact Upload"
+
+inputs:
+  build_directory:
+    type: string
+  amdgpu_families:
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: Configure AWS Credentials
+      if: always()
+      uses: aws-actions/configure-aws-credentials@ececac1a45f3b08a01d2dd070d28d111c5fe6722 # v4.1.0
+      with:
+        aws-region: us-east-2
+        role-to-assume: arn:aws:iam::692859939525:role/therock-artifacts
+
+    - name: Create Logs index Files
+      shell: bash
+      if: always()
+      run: |
+        python3 build_tools/github_actions/create_log_index.py \
+          --build-dir=${{ inputs.build_directory }} \
+          --amdgpu-family=${{ inputs.amdgpu_families }}
+
+    - name: Upload artifacts
+      shell: bash
+      if: always()
+      run: |
+        python build_tools/github_actions/upload_build_artifacts.py \
+          --run-id ${{ github.run_id }} \
+          --amdgpu-family ${{ inputs.amdgpu_families }} \
+          --build-dir ${{ inputs.build_directory }}
+
+    - name: Upload Logs
+      shell: bash
+      if: always()
+      run: |
+        python3 build_tools/github_actions/upload_build_logs_to_s3.py \
+          --build-dir=${{ inputs.build_directory }} \
+          --run-id ${{ github.run_id }} \
+          --amdgpu-family ${{ inputs.amdgpu_families }}
```

Comment:
(Not in this PR, but relating to the positioning of this as a composite workflow)

This should be moved to teatime.py: https://github.com/ROCm/TheRock/pull/648